### ISSUE TITLE
c-mode comment deletes new line between successive comments

### ISSUE THREAD

**Issue by broncodev** [joaotavora/yasnippet#791]
Hello.
First of all, thank you for yasnippet. I would probably not use emacs without it.

I'm having a some-what strange behaviour when I try to create a comment preceding another comment (two successive lines; the first line is initially empty and the second line already has a comment).

To reproduce:
1. Create a new buffer with c-mode: `C-x C-f test.c`
2. Activate c-mode if not already active: `M-x c-mode`
3. Type `co`
4. Complete snippet: `C-S-i`
5. Type anything that you like
6. Hit tab
7. Go to the beginning of line: `C-a`
8. Open a new line: `C-o`
9. Type `co`
10. Complete snippet: `C-S-i`
11. Type anything that you like with a space

Bug:
when you hit the space bar, the new line between the two comments is deleted, and they end up in the same line, separated by ` */`.

My configuration (running emacs -Q):
emacs GNU Emacs 24.4.1 (x86_64-pc-linux-gnu, GTK+ Version 3.4.2)
yasnippet (version 0.11.0) -- pluskid/joaotavora/npostavs

.emacs (init file)
```
(require 'package)
(package-initialize)
(require 'yasnippet)
(define-key yas-minor-mode-map (kbd "<tab>") nil)
(define-key yas-minor-mode-map (kbd "TAB") nil)
(define-key yas-minor-mode-map (kbd "C-i") nil)
(define-key yas-minor-mode-map (kbd "C-S-i") 'yas-expand)

(add-hook 'prog-mode-hook 'yas-minor-mode)

(require 'function-args)
(add-hook 'c-mode-common-hook (lambda ()
                                (function-args-mode) ; Find definitions in current buffer
                                (setq tab-width 4) ; set tab width to 4
                                (auto-fill-mode) ; break long lines
                                (yas-minor-mode) ; activate yasnippet
                                ))
```
May be related to https://github.com/joaotavora/yasnippet/issues/784

**Comment by npostavs** [comment]
> May be related to #784

I think it's probably another manifestation of #784, at least, I can't seem to reproduce when using the latest yasnippet HEAD (b4d2f9cb4e3ec63b36213fffc10d9591b35b8bfa). If you are installing from MELPA, it will probably take another 6-8 hours before the fix arrives.

**Comment by broncodev** [comment]
Fixed, indeed (yasnippet-20170326.1030)
